pandas/io/tests/test_parsers.py

```diff
@@ -4652,6 +4655,9 @@ def test_invalid_c_parser_opts_with_not_c_parser(self):
         engines = 'python', 'python-fwf'
         for default in c_defaults:
             for engine in engines:
+                if 'python' in engine and default not in py_unsupported:
```

Comment:
The test name indicates that the answer is no.  However, I do believe that the invalid parameters are tested but in different parts of the test suite (again, this is where a refactor would be good in a separate PR).